language/rescue_spec.rb

```diff
@@ -223,14 +285,39 @@ class ArbitraryException < StandardError
     a.should == 'ac'
   end
 
-  it "without classes will not rescue Exception" do
-    lambda do
+  context "without rescue expression" do
+    it "will rescue only StandardError and its subclasses" do
       begin
-        raise Exception
+        raise StandardError
       rescue
-        'Exception wrongly rescued'
+        ScratchPad << :caught
       end
-    end.should raise_error(Exception)
+
+      ScratchPad.recorded.should == [:caught]
+    end
+
+    it "will not rescue exceptions except StandardError" do
+      [Exception, NoMemoryError, ScriptError, SecurityError, SystemExit, SystemStackError].each do |klass|
+        lambda {
+          begin
+            raise klass
+          rescue
+            ScratchPad << :caught
+          end
+        }.should raise_error(klass)
+      end
+      ScratchPad.recorded.should == []
+    end
+
+    it "will still rescue SignalException" do
+      begin
+        raise SignalException
+      rescue
+        ScratchPad << :caught
+      end
+
+      ScratchPad.recorded.should == [:caught]
+    end
```

Comment:
again, looks like a bug

Reply:
```ruby
> begin; raise SignalException; rescue => e; p e; end
#<ArgumentError: wrong number of arguments (given 0, expected 1)>
```
So SignalException expects an argument, like `"INT"`.

Reply:
It's the tricky problem that `raise` itself or the Exception's constructor can raise exceptions.